core/cli/src/lib.rs

```diff
@@ -419,7 +421,13 @@ impl<'a> ParseAndPreparePurge<'a> {
 		let config = create_config_with_db_path::<(), _, _, _>(
 			spec_factory, &self.params.shared_params, self.version
 		)?;
-		let db_path = config.database_path;
+		let db_path = match config.database {
+			ConfigurationDb::Path { path, .. } => path,
+			_ => {
+				println!("Cannot purge custom database implementation");
```

Comment:
I'd use `error!` for this

Reply:
I went for `println!` because this is a CLI function, and the rest of the function also uses `println!` several times.

Reply:
Normally errors should be printed to stderr.